### PR TITLE
Syndication Blogs API Endpoint and API Key Auth (LBSD-1049, LBSD-1041)

### PR DESCRIPTION
**Pull request by mscam**
- LBSD-1041: Syndication Blogs API Endpoint
- LBSD-1049: Syndication API Key Auth
- Behave tests & utilities for Consumer & API Key Auth

### CODE REVIEW

File: server/features/syndication_blogs.feature [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            {
+                "title": "testBlog3",
+                "blog_status": "open",
+                "syndication_enabled": true,
+                "blog_preferences": {"theme": "forest", "language": "fr"}
+            }
+        ]
+        """
+        Given config consumer api_key
+        When we get "/syndication/blogs/"
```
Comment (m0g) [inline server/features/syndication_blogs.feature]: Wouldn't it make more sense to have an endpoint like `/producers/:id/blogs` ?

**Comment by mscam** [comment]
@liveblog/liveblog-devs please have a look at this PR. Thank you!

**Comment by nistormihai** [comment]
LGTM

File: server/liveblog/tests/environment.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+from apps.io.tests import setup_providers
+from superdesk import tests
+from superdesk.tests import setup_auth_user
+from superdesk.tests.environment import setup_search_provider
+from superdesk.vocabularies.command import VocabulariesPopulateCommand
+from liveblog.tests import setup_auth_consumer, test_consumer
+
+
+def setup_before_scenario(context, scenario, config, app_factory):
```
Comment (naspeh) [inline server/liveblog/tests/environment.py]: Could you use `setup_before_scenario` from `superdesk` and add your modifications without duplicate all other general parts

Comment (mscam) [inline server/liveblog/tests/environment.py]: I think that is better to have our environment.py and remove dependency with Superdesk tests. If you want to improve the Superdesk project with our changes instead, we can discuss about that, but I don't think that you need Syndication testing environment and scenarios.

**Pull request merged**